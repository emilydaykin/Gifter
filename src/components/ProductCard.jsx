const ProductCard = ({ product }) => {
  return (
    <div className='product-card'>
      <img className='product-card__image' src={product.imageUrl} alt={product.name} />
      <button className='button product-card__button'>Add to Card</button>
      <div className='product-card__product-info'>
        <h3 className='product-card__product-name'>{product.name}</h3>
        <p className='product-card__product-price'>&euro;{product.price}</p>
      </div>
    </div>
  );
};

export default ProductCard;
