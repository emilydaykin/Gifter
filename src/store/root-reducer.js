import { combineReducers } from 'redux'; // to create a big final one
import { userReducer } from './user/user.reducer';
import { categoriesReducer } from './categories/category.reducer';
import { cartReducer } from './cart/cart.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  categories: categoriesReducer,
  cart: cartReducer
});
