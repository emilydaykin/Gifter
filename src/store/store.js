import { compose, createStore, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // this will be by default local storage in most web browsers
import { rootReducer } from './root-reducer';
import logger from 'redux-logger';
// import { loggerMiddleware } from './middleware/logger';
import thunk from 'redux-thunk';

const persistConfig = {
  key: 'root',
  storage,
  // blacklist: ['user'] // blacklist the user reducer (since it comes from the auth state listener anyway)
  whitelist: ['cart']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

// Only show logs in dev, not prod!
const middlewares = [process.env.NODE_ENV !== 'production' && logger, thunk].filter(Boolean); // enhance our store
// const middlewares = [process.env.NODE_ENV === 'development' && loggerMiddleware].filter(Boolean); // enhance our store

// Allowing Chrome to use Redux Dev Tools if Chrome extension is installed, otherwise use Redux's compose:
const composeEnhancer =
  (process.env.NODE_ENV !== 'production' &&
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

const composedEnhancers = composeEnhancer(applyMiddleware(...middlewares));

// root-reducer
export const store = createStore(persistedReducer, undefined, composedEnhancers);

export const persistor = persistStore(store);
