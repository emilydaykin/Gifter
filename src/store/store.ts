import { compose, createStore, applyMiddleware, Middleware } from 'redux';
import { persistStore, persistReducer, PersistConfig } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // this will be by default local storage in most web browsers
import { rootReducer } from './root-reducer';
import logger from 'redux-logger';
// import { loggerMiddleware } from './middleware/logger';
// import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './root-saga';

export type RootState = ReturnType<typeof rootReducer>;

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

type ExtendedPersistConfig = PersistConfig<RootState> & {
  whitelist: (keyof RootState)[];
};

const persistConfig: ExtendedPersistConfig = {
  key: 'root',
  storage,
  // blacklist: ['user'] // blacklist the user reducer (since it comes from the auth state listener anyway)
  whitelist: ['cart']
};

const sagaMiddleware = createSagaMiddleware();

const persistedReducer = persistReducer(persistConfig, rootReducer);

// Only show logs in dev, not prod!
const middlewares = [process.env.NODE_ENV !== 'production' && logger, sagaMiddleware].filter(
  (middleware): middleware is Middleware => Boolean(middleware)
); // enhance our store
// const middlewares = [process.env.NODE_ENV !== 'production' && loggerMiddleware].filter(Boolean); // enhance our store

// Allowing Chrome to use Redux Dev Tools if Chrome extension is installed, otherwise use Redux's compose:
const composeEnhancer =
  (process.env.NODE_ENV !== 'production' &&
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

const composedEnhancers = composeEnhancer(applyMiddleware(...middlewares));

// root-reducer
export const store = createStore(persistedReducer, undefined, composedEnhancers);

sagaMiddleware.run(rootSaga);

export const persistor = persistStore(store);
