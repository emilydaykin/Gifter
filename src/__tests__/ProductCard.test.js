import { render } from '@testing-library/react';
import ProductCard from '../components/ProductCard';
import { Provider } from 'react-redux';
import { store } from '../store/store';

const mockProduct = {
  id: 25,
  name: 'Garden Book Shelf Box',
  imageUrl:
    'https://images.unsplash.com/photo-1591492654773-6756035bef6b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8c2VsZWN0aW9uJTIwbm9uJTIwZmljdGlvbiUyMGJvb2tzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
  price: 87.5
};

test('Product Card in Shop preview renders', () => {
  const view = render(
    <Provider store={store}>
      <ProductCard product={mockProduct} preview={true} />
    </Provider>
  );

  expect.assertions(1);

  expect(view).toMatchSnapshot();
});

test('Product Card in specific shop category renders', () => {
  const view = render(
    <Provider store={store}>
      <ProductCard product={mockProduct} preview={false} />
    </Provider>
  );

  expect.assertions(1);

  expect(view).toMatchSnapshot();
});
