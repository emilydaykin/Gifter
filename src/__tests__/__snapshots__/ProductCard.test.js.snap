// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card in Shop preview renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-card product-card--preview"
      >
        <img
          alt="Garden Book Shelf Box"
          class="product-card__image"
          src="https://images.unsplash.com/photo-1591492654773-6756035bef6b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8c2VsZWN0aW9uJTIwbm9uJTIwZmljdGlvbiUyMGJvb2tzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60"
        />
        <button
          class="button product-card__button"
        >
          Add to Cart
        </button>
        <div
          class="product-card__product-info"
        >
          <h3
            class="product-card__product-name"
          >
            Garden Book Shelf Box
          </h3>
          <p
            class="product-card__product-price"
          >
            €
            87.50
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-card product-card--preview"
    >
      <img
        alt="Garden Book Shelf Box"
        class="product-card__image"
        src="https://images.unsplash.com/photo-1591492654773-6756035bef6b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8c2VsZWN0aW9uJTIwbm9uJTIwZmljdGlvbiUyMGJvb2tzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60"
      />
      <button
        class="button product-card__button"
      >
        Add to Cart
      </button>
      <div
        class="product-card__product-info"
      >
        <h3
          class="product-card__product-name"
        >
          Garden Book Shelf Box
        </h3>
        <p
          class="product-card__product-price"
        >
          €
          87.50
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product Card in specific shop category renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-card"
      >
        <img
          alt="Garden Book Shelf Box"
          class="product-card__image"
          src="https://images.unsplash.com/photo-1591492654773-6756035bef6b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8c2VsZWN0aW9uJTIwbm9uJTIwZmljdGlvbiUyMGJvb2tzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60"
        />
        <button
          class="button product-card__button"
        >
          Add to Cart
        </button>
        <div
          class="product-card__product-info"
        >
          <h3
            class="product-card__product-name"
          >
            Garden Book Shelf Box
          </h3>
          <p
            class="product-card__product-price"
          >
            €
            87.50
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-card"
    >
      <img
        alt="Garden Book Shelf Box"
        class="product-card__image"
        src="https://images.unsplash.com/photo-1591492654773-6756035bef6b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8c2VsZWN0aW9uJTIwbm9uJTIwZmljdGlvbiUyMGJvb2tzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60"
      />
      <button
        class="button product-card__button"
      >
        Add to Cart
      </button>
      <div
        class="product-card__product-info"
      >
        <h3
          class="product-card__product-name"
        >
          Garden Book Shelf Box
        </h3>
        <p
          class="product-card__product-price"
        >
          €
          87.50
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
